### PR TITLE
Enable flake8 pytest style

### LINKED ISSUE #9028: Enable rules in `flake8-pytest-style`
Enable the following rules in https://beta.ruff.rs/docs/rules/#flake8-pytest-style-pt:

- [x] PT001
- [x] PT006
- [x] PT007
- [x] PT009
- [x] PT010
- [ ] PT012
- [x] PT018
- [x] PT023
- [x] PT026

## Settings

```toml
[tool.ruff.flake8-pytest-style]
mark-parentheses = false
fixture-parentheses = false
```

### PR DESCRIPTION
**Pull request by EdAbati**
## Related Issues/PRs

Towards #9028 

## What changes are proposed in this pull request?

I have enabled the rules one by one in each commit.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@EdAbati Thanks for the PR!

> I have enabled the rules one by one in each commit.

Sorry, I wasn't clear. What I meant was enable the rules one-PR by one-PR. More code = More review cost = Higher chances to miss something. For example, a PR to only enable PT001 is easier to review and has lower chances to miss invalid changes than this PR.

**Comment by EdAbati** [comment]
Sounds good I'll do one for each rule

**Comment by harupy** [comment]
@EdAbati Great! You can enable the rules that require any fixes in a single PR, but for ones that require fixes, let's do one for each rule.

**Comment by EdAbati** [comment]
Ah ok! FYI in this PR I added for now only the Ruff codes that have automatic fixes.

Should I leave it like this then?

**Comment by harupy** [comment]
@EdAbati Got it, let's merge this PR :)

File: pyproject.toml [hunk 1]
```diff
@@ -73,3 +82,6 @@ exclude = [
 ]
 [tool.ruff.per-file-ignores]
 "dev/*" = ["T201"]
+[tool.ruff.flake8-pytest-style]
```
Comment (harupy) [inline pyproject.toml:85]:
```suggestion

[tool.ruff.flake8-pytest-style]
```

File: pyproject.toml [hunk 2]
```diff
@@ -18,6 +18,15 @@ select = [
   "E",
   "F",
   "C4",
+  "PT001",
+  "PT006",
+  # "PT007",
```
Comment (harupy) [inline pyproject.toml:23]: Can we add a TODO comment here and say "Enable commented out rules?"

**Review by harupy — APPROVED** [review]
@EdAbati Sorry for the delay, LGTM!

**Comment by harupy** [comment]
@EdAbati Can you rebase this PR on master? There might be changes violating the new rules on master branch.

**Comment by EdAbati** [comment]
Hi @harupy , it should be updated now :)

**Pull request merged**